Health goals integration

ryan-txn opened the pull request.

ryan-txn committed:
Rename Intendation as Indentation

ryan-txn committed:
Made Indentation public for usage in ChatParser askForHealthGoal()

ryan-txn committed:
Add asking for health goal

Ask for weight, height and health goal at beginning of run() method. Incomplete parseHealthGoal method with switch cases

ryan-txn committed:
Add set health goal to list of commands

ryan-txn committed:
Rename health goals

ryan-txn committed:
User class clean up

Make existing variables and methods static, include healthgoal and idealcalories variables, correct toString() method previously not zero-based

kennethSty reviewed (changes requested):
Hi! Please fix these checkstyle issues:

1.  /home/runner/work/tp/tp/src/main/java/seedu/healthmate/ChatParser.java:55: Line is longer than 120 characters (found 122). [LineLength]
2.  /home/runner/work/tp/tp/src/main/java/seedu/healthmate/UI.java:6:5: Variable access definition in wrong order. [DeclarationOrder]

kennethSty commented on src/main/java/seedu/healthmate/ChatParser.java:
```diff
@@ -79,4 +101,17 @@ public void multiCommandParsing(String userInput) {
         }
     }
 
+    public void parseHealthGoal(String healthGoal) {
```
To keep the ChatParser class clean, consider creating a Healthgoal class to encapsulate all related methods such as `parseHealthGoal`.

kennethSty commented on src/main/java/seedu/healthmate/User.java:
```diff
@@ -2,26 +2,29 @@
     public User() {
-        this.weightEntries = new ArrayList<>();
-        this.heightEntries = new ArrayList<>();
+        weightEntries = new ArrayList<>();
```
This class is a bit confusing to me. Why the changes to make everything static?

kennethSty commented on src/main/java/seedu/healthmate/UI.java:
```diff
@@ -3,41 +3,42 @@
 
     private static final String SEPARATOR = "_________________________________________________________________________";
-    private static final String INTENDATION = "      ";
+    public static final String INDENTATION = "      ";
```
Great catch! Thanks!

kennethSty commented on src/main/java/seedu/healthmate/ChatParser.java:
```diff
@@ -27,6 +27,8 @@ public ChatParser(){
      * and initiates the parsing process steered by one-token and two-token-based user prompts.
      */
     public void run() {
+        askForHealthGoal();
```
Great! 
Could you please create a new Issue for further developing this? Specifically, to make the execution of askForHealthGoal optional, such that the user is not asked to insert a HealthGoal every time he opens the app.

ryan-txn committed:
Update test file for asking health goal

ryan-txn committed:
Merge branch 'master' into Health-Goals_Integration

ryan-txn committed:
Resolve Merge Conflict

resolve INDENTATION naming merge conflict

ryan-txn committed:
Resolve merge conflicts 2

Rename INDENTATION again

ryan-txn committed:
Solve CI test issues

1. split line longer than 120 characters
2. reorder public variable to be on top

ryan-txn committed:
Fix expected output

Remove unneeded spaces

ryan-txn committed:
Fix expected file 2

Include space in expected for new line

ryan-txn committed:
Revert "Fix expected file 2"

This reverts commit 3b25582ed395379f3426f497390660c75feb3795.

kennethSty commented on src/main/java/seedu/healthmate/UI.java:
```diff
@@ -2,51 +2,56 @@
 
 public class UI {
 
+    public static final String INDENTATION = "      ";
```
Please do not make private members public.

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration

# Conflicts:
#	src/main/java/seedu/healthmate/ChatParser.java
#	src/main/java/seedu/healthmate/UI.java

ryan-txn committed:
Revert "Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration"

This reverts commit 7f566bd6e6e60bf9c6fba2f99fdb2ad1922a2f87, reversing
changes made to 27cf870945c78c626b2aaf0564eba781627186cc.

ryan-txn committed:
Reapply "Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration"

This reverts commit f746b252a8ac2fced4dc20e14f74a4efb05fde72.

ryan-txn committed:
Reorder static variable for checkstyle

ryan-txn committed:
remove buggy method

extra semicolon

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration

# Conflicts:
#	src/main/java/seedu/healthmate/UI.java

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration

# Conflicts:
#	src/main/java/seedu/healthmate/Commands.java
#	src/main/java/seedu/healthmate/UI.java

ryan-txn committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-1/tp into Health-Goals_Integration

# Conflicts:
#	src/main/java/seedu/healthmate/HealthGoal.java
#	src/main/java/seedu/healthmate/User.java

ryan-txn committed:
Add user data file

ryan-txn committed:
Fix CI test issues

ryan-txn committed:
Fix spacing issue

kennethSty reviewed (approved):
Great changes!

kennethSty commented:
Modifications made in #69
